### ISSUE TITLE
Yarn does not resolve GIT URLs with the commit-ish format of #semver:<semver>

### ISSUE THREAD

**Issue by riebel** [yarnpkg/yarn#5342]
**Do you want to request a *feature* or report a *bug*?**

feature / inconsistency with npm 5.6.0

**What is the current behavior?**

A dependency of the format:
```
"dependencies": {
     "company-bundle": "git+ssh://git@bitbucket.org/company/bundle.git#semver:^1.0.0"
}
```
cannot be resolved:

`error Couldn't find match for "semver:^1.0.0" in "_c" for "ssh://git@bitbucket.org/company/bundle.git".`

**What is the expected behavior?**

yarn should resolve the URL like npm 5.6.0

[ If the commit-ish has the format #semver:<semver>, <semver> can be any valid semver range or exact version, and npm will look for any tags or refs matching that range in the remote repository, much as it would for a registry dependency.](https://docs.npmjs.com/files/package.json#git-urls-as-dependencies)

**Please mention your node.js, yarn and operating system version.**

node v9.3.0
yarn 1.3.2
npm 5.6.0
Windows 10 Professional

**Comment by elpdrc** [comment]
@riebel This is happen on windows only?

**Comment by riebel** [comment]
no this was a general issue on all platforms. But it seems to be fixed in yarn 1.4.1 #5126 #5140
